src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/ConfigurationManagerExtensions.cs

```diff
@@ -4,13 +4,21 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Options;
+using System.IO;
+using System;
 
 namespace CoreWCF.Configuration
 {
     public static class ConfigurationManagerExtensions
     {
         public static IServiceCollection AddServiceModelConfigurationManagerFile(this IServiceCollection builder, string path)
         {           
+            if (!File.Exists(path)) { throw new FileNotFoundException(SR.Format(SR.FileNotFound, path)); }
+            using (var fs = new FileStream(path, FileMode.Open))
```

Comment:
```suggestion
            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read))
```
 The tests use `TemporaryFileStream` which opens the file with the default sharing mode of `FileShare.Read`. As `Write` wasn't specified in the `FileShare` parameter, it denies subsequently opening the file for write (as long as it's still open by `TemporaryFileStream`). Your code wasn't specifying any `FileAccess` mode, so it requested the default of `ReadWrite`, which gets denied because `TemporaryFileStream` is still holding a file handle which denies opening the file in write mode.  

This small change should fix the tests.